### PR TITLE
feat: support custom time stamp in story cards

### PR DESCRIPTION
**Pull request by snitin4260**
# Description
fixes https://github.com/quintype/page-builder/issues/2137

Please include the summary of the change made. Please include the required context here. Please include the issue reference for the fix and the dependencies reference for the change.

Fixes # (issue-reference)

Dependencies # (dependency-issue-reference)

Documentation # (link to the corresponding documentation changes)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test Case A
- [ ] Test Case B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/molecules/related-story-card/related-story-card.tsx [hunk 1]
```diff
@@ -32,10 +32,16 @@ export const RelatedStoryCard = ({ story, config }: RelatedStoryCardTypes) => {
 
   const languageCode = get(config, ["publisherConfig", "language", "ietf-code"]);
   const localizationOpts = get(config, ["opts", "featureConfig", "localization"], {});
+  let timeStamp = lastPublishedAt;
+  const storyCardTime = get(config, ["opts", "featureConfig", "storyCardTime"], null);
+  const customTimeStamp = storyCardTime && typeof storyCardTime === "function" && storyCardTime(config, story);
+  if (customTimeStamp) {
+    timeStamp = customTimeStamp;
+  }
```
Comment (ags1773) [inline src/molecules/related-story-card/related-story-card.tsx:40]:
```suggestion
  if (storyCardTime && typeof storyCardTime === "function") {
    timeStamp = storyCardTime(config, story);
  }
```

Comment (snitin4260) [reply src/molecules/related-story-card/related-story-card.tsx:40]: @ags1773 We don't know if storyCardTime(config, story) is truthy in this version

Comment (ags1773) [reply src/molecules/related-story-card/related-story-card.tsx:40]: It doesn't matter.. we can assume that the function returns a proper value

Comment (Jeevan-Kishore) [reply src/molecules/related-story-card/related-story-card.tsx:40]: +1

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]

**Review by ags1773 — APPROVED** [review]

**Review by ags1773 — APPROVED** [review]

**Review by Athira001 — APPROVED** [review]

**Review by Jeevan-Kishore — APPROVED** [review]

**Pull request merged**